cpp/arcticdb/python/python_handlers.cpp

```diff
@@ -92,51 +69,113 @@ void EmptyHandler::handle_type(
+    }
 }
 
 int EmptyHandler::type_size() const {
     return sizeof(PyObject *);
 }
 
-void EmptyHandler::default_initialize(void *dest, size_t byte_size) const {
-    fill_with_none(reinterpret_cast<const PyObject**>(dest), byte_size / type_size());
+void EmptyHandler::default_initialize(ChunkedBuffer& buffer, size_t bytes_offset, size_t byte_size, const DecodePathData&, std::any& any) const {
+    auto& handler_data = get_handler_data(any);
+    fill_with_none(buffer, bytes_offset, byte_size / type_size(), handler_data.spin_lock());
 }
 
 void BoolHandler::handle_type(
-    const uint8_t *&data,
-    uint8_t *dest,
-    const EncodedFieldImpl &field,
-    const ColumnMapping& m,
-    size_t,
-    const std::shared_ptr<BufferHolder>&,
-    EncodingVersion encoding_version) {
+        const uint8_t *&data,
+        ChunkedBuffer& dest_buffer,
+        const EncodedFieldImpl &field,
+        const ColumnMapping& m,
+        const DecodePathData& shared_data,
+        std::any& handler_data,
+        EncodingVersion encoding_version,
+        const std::shared_ptr<StringPool>&) {
     ARCTICDB_SAMPLE(HandleBool, 0)
-    util::check(dest != nullptr, "Got null destination pointer");
     util::check(field.has_ndarray(), "Bool handler expected array");
     ARCTICDB_DEBUG(log::version(), "Bool handler got encoded field: {}", field.DebugString());
-    auto ptr_dest = reinterpret_cast<const PyObject**>(dest);
     const auto &ndarray = field.ndarray();
     const auto bytes = encoding_sizes::data_uncompressed_size(ndarray);
-    ChunkedBuffer decoded_data = ChunkedBuffer::presized(bytes);
-    SliceDataSink decoded_data_sink{decoded_data.data(), bytes};
-    std::optional<util::BitSet> sparse_map;
-    data += decode_field(m.source_type_desc_, field, data, decoded_data_sink, sparse_map, encoding_version);
-    const auto num_bools = sparse_map.has_value() ? sparse_map->count() : m.num_rows_;
-    auto ptr_src = decoded_data.template ptr_cast<uint8_t>(0, num_bools * sizeof(uint8_t));
+    Column decoded_data = Column(m.source_type_desc_, bytes / get_type_size(m.source_type_desc_.data_type()));
+    data += decode_field(m.source_type_desc_, field, data, decoded_data, decoded_data.opt_sparse_map(), encoding_version);
+
+    convert_type(
+        decoded_data,
+        dest_buffer,
+        m.num_rows_,
+        m.offset_bytes_,
+        m.source_type_desc_,
+        m.dest_type_desc_,
+        shared_data,
+        handler_data,
+        {});
+}
+
+void BoolHandler::convert_type(
+        const Column& source_column,
+        ChunkedBuffer& dest_buffer,
+        size_t num_rows,
+        size_t offset_bytes,
+        arcticdb::entity::TypeDescriptor,
+        arcticdb::entity::TypeDescriptor,
+        const arcticdb::DecodePathData &,
+        std::any& any,
+        const std::shared_ptr<StringPool> &) {
+    const auto& sparse_map = source_column.opt_sparse_map();
+    const auto num_bools = sparse_map.has_value() ? sparse_map->count() : num_rows;
```

Comment:
Do we ever write columns with no values? So sparse map doesn't have a value, but there is no data either?

Reply:
Not at the moment, we will write empty-type columns but they should never get here.